src/lib.rs

```diff
@@ -102,6 +142,49 @@ impl SynapseVersion {
     }
 }
 
+/// The configuration for a single synapse container.
+/// We copy some fields from the main config because they
+/// are otherwise without the context we want to use them for, which
+/// is managing the docker container, not deserializing a homserver config.
```

Comment:
I don't understand. How is that a "context"? That looks like an implementation detail, no?

Reply:
Yeah it is an implementation detail but I wanted to explain why I did it because the alternative is horrible.